[Bots] Correct ^pull logic and add checks for Enchanter pets

nytmyr opened the pull request:
# Description

- Removed initial LoS check from bots and rely only on owner when told to ^pull. This prevents bots from getting stuck in a loop if they don't have LoS when trying to initiate the pull.
- Fixed Animation Empathy checks and consolidates.
- Enchanter pets will honor Animation Empathy rules all around.
- Pets will properly guard if allowed when pulling.
- Adds additional checks to ensure pulls go properly and abort properly.
- Uses tiered checks when within range to do the appropriate attack (Archery->Spell AI->Pull Spell [Throw Stone by default])

[Bot Pull Bug](https://discord.com/channels/212663220849213441/1350097470620831794/1355671467404300288)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Testing

- Verified LoS scenarios and different types of pulling with bots.
- Verified Enchanter pets are only controllable when applicable (proper ranked of Animation Empathy).
- Verified Pet guards properly on pulls and releases guard on return.

Clients tested: RoF2

# Checklist

- [x] I have tested my changes
- [x] I have performed a self-review of my code. Ensuring variables, functions and methods are named in a human-readable way, comments are added only where naming of variables, functions and methods can't give enough context.
- [x] I own the changes of my code and take responsibility for the potential issues that occur

nytmyr committed:
[Bots] Fix bots getting stuck on ^pull

- Removed initial LoS check from bots and rely only on owner when told to ^pull. This prevents bots from getting stuck in a loop if they don't have LoS when trying to initiate the pull.

nytmyr committed:
[Bots] Add valid state checks to ^clickitem

- Bots previously weren't checking if they were in a valid state or not before trying to click an item when told (in a death state, held, feared, silenced, in group/raid, etc.)

nytmyr committed:
[Bots] Add valid state checks to ^clickitem

- Bots previously weren't checking if they were in a valid state or not before trying to click an item when told (in a death state, held, feared, silenced, in group/raid, etc.)

Miscommit

nytmyr committed:
Merge branch 'bot-pull-fix' of https://github.com/nytmyr/Server into bot-pull-fix

nytmyr committed:
undo

Akkadius commented on zone/bot.cpp:
```diff
@@ -3564,7 +3561,14 @@ void Bot::BotPullerProcess(Client* bot_owner, Raid* raid) {
 			bot_owner->SetBotPulling();
 
-			if (HasPet() && (GetClass() != Class::Enchanter || GetPet()->GetPetType() != petAnimation || GetAA(aaAnimationEmpathy) >= 1)) {
+			if (
+				HasPet() &&
+					(
+						GetClass() != Class::Enchanter ||
+						GetPet()->GetPetType() != petAnimation ||
+						GetAA(aaAnimationEmpathy) >= 1
+					)
+			) {
```
Dear lord

Akkadius replied:
Can we break this out into readable bools

nytmyr committed:
Cleanup and Fixes

- Cleanup
- Fixed Animation Empathy checks and consolidates
- Enchanter pets will honor Animation Empathy rules all around
- Pets will properly guard if allowed when pulling

nytmyr committed:
Merge branch 'master' into bot-pull-fix
